module/zfs/zfs_ioctl.c

```diff
@@ -5039,17 +5039,13 @@ zfs_ioc_pool_reopen(zfs_cmd_t *zc)
 	spa_t *spa;
 	int error;
 	nvlist_t *reopenflags;
-	boolean_t no_scrub_restart;
+	boolean_t scrub_restart = B_TRUE;
 
 	if (zc->zc_nvlist_src != 0 &&
 	    (error = get_nvlist(zc->zc_nvlist_src, zc->zc_nvlist_src_size,
-	    zc->zc_iflags, &reopenflags)) != 0)
-		return (error);
-
-	error = nvlist_lookup_boolean_value(reopenflags,
-	    "no_scrub_restart", &no_scrub_restart);
-	if (error != 0)
-		return (SET_ERROR(EINVAL));
+	    zc->zc_iflags, &reopenflags)) == 0)
+		nvlist_lookup_boolean_value(reopenflags,
+		    "scrub_restart", &scrub_restart);
```

Comment:
There's a leak here, `reopenflags` must be freed.  Or we need to switch `zfs_ioc_pool_reopen` to the new style ioctl's.